diff --git a/mc/include/alps/mc/mpiadapter.hpp b/mc/include/alps/mc/mpiadapter.hpp
index 23df04cd..5890d56f 100644
--- a/mc/include/alps/mc/mpiadapter.hpp
+++ b/mc/include/alps/mc/mpiadapter.hpp
@@ -219,8 +219,13 @@ namespace alps {
         static parameters_type& define_parameters(parameters_type & parameters) {
             base_type_::define_parameters(parameters);
             if (parameters.is_restored()) return parameters;
-            parameters.template define<std::size_t>("Tmin", 1, "minimum time to check if simulation has finished");
-            parameters.template define<std::size_t>("Tmax", 600, "maximum time to check if simulation has finished");
+#if defined(_WIN32)
+            using parameter_size_type = unsigned long;
+#else
+            using parameter_size_type = std::size_t;
+#endif
+            parameters.template define<parameter_size_type>("Tmin", 1, "minimum time to check if simulation has finished");
+            parameters.template define<parameter_size_type>("Tmax", 600, "maximum time to check if simulation has finished");
             return parameters;
         }
     };
diff --git a/params/include/alps/params/convenience_params.hpp b/params/include/alps/params/convenience_params.hpp
index 7b22d93e..3a5c23e6 100644
--- a/params/include/alps/params/convenience_params.hpp
+++ b/params/include/alps/params/convenience_params.hpp
@@ -15,10 +15,15 @@ namespace alps {
         /// @brief Defines a number of frequently-used parameters.
         /// Defines `size_t timelimt`, `string outputfile`, `string checkpoint`.
         inline params& define_convenience_parameters(params & parameters) {
+#if defined(_WIN32)
+            using parameter_size_type = unsigned long;
+#else
+            using parameter_size_type = std::size_t;
+#endif
             const std::string origin=origin_name(parameters);
             const std::string basename=alps::fs::remove_extensions(alps::fs::get_basename(origin));
             parameters
-                .define<std::size_t>("timelimit", 0, "time limit for the simulation")
+                .define<parameter_size_type>("timelimit", 0, "time limit for the simulation")
                 .define<std::string>("outputfile", basename+".out.h5", "name of the output file")
                 ;
             const std::string base_outname=alps::fs::remove_extensions(alps::fs::get_basename(parameters["outputfile"]));
